### PR TITLE
Add forgot password and verification email template

### LINKED ISSUE #876: Create app settings to configure e-mail templates for validation and password reset
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26911882-create-app-settings-to-configure-e-mail-templates-for-validation-and-password-reset?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by snaiperskaya96**
@RocketChat/core 

Closes #876

Adds templates for the verification and forgot password emails
screen shot 2017-01-16 at 3 49 06 pm

### CODE REVIEW

File: server/methods/registerUser.coffee [hunk 1]
```diff
@@ -29,6 +29,13 @@ Meteor.methods
 
 		try
 			if userData.email
+				if RocketChat.settings.get('Verification_Customized')
+					subject = RocketChat.placeholders.replace(RocketChat.settings.get('Verification_Email_Subject') || '')
+					html = RocketChat.placeholders.replace(RocketChat.settings.get('Verification_Email') || '')
+					Accounts.emailTemplates.verifyEmail.subject = (userModel) ->
+						return subject
+					Accounts.emailTemplates.verifyEmail.html = (userModel, url) ->
+						return html.replace('[Verification_Url]', url);
```
Comment (graywolf336) [inline server/methods/registerUser.coffee]: All of the `.replace(String, 'something')` will only replace it once. This needs to be converted into regex for it to replace every instance of that string. So, in this case `html.replace(/\[Verification_Url\]/g, url)`. Just thinking that someone could put that in a url and then below put it again if a client can only read the text and not the anchors.

Comment (snaiperskaya96) [inline server/methods/registerUser.coffee]: That's right! Gonna edit it in an hour

Comment (graywolf336) [inline server/methods/registerUser.coffee]: :+1:

Comment (snaiperskaya96) [inline server/methods/registerUser.coffee]: Done! 👍

**Comment by snaiperskaya96** [comment]
Any idea about the last travis fail? I mean what's the point of failing by changing from `return html.replace('[Verification_Url]', url);` to `return html.replace(/\[Verification_Url]/g, url);`? (That's the only change i've done in the last commit - the one that fails)

**Comment by graywolf336** [comment]
The error that occurred is a false positive, the build just needs to be triggered again.

**Comment by engelgabriel** [comment]
retrying now

File: server/methods/sendConfirmationEmail.coffee [hunk 2]
```diff
@@ -6,6 +6,14 @@ Meteor.methods
 		user = RocketChat.models.Users.findOneByEmailAddress s.trim(email)
 
 		if user?
+			if RocketChat.settings.get('Verification_Customized')
+				subject = RocketChat.placeholders.replace(RocketChat.settings.get('Verification_Email_Subject') || '')
+				html = RocketChat.placeholders.replace(RocketChat.settings.get('Verification_Email') || '')
+				Accounts.emailTemplates.verifyEmail.subject = (userModel) ->
+					return subject
+				Accounts.emailTemplates.verifyEmail.html = (userModel, url) ->
+					return html.replace('[Verification_Url]', url);
```
Comment (graywolf336) [inline server/methods/sendConfirmationEmail.coffee]: You forgot one :thinking:

Comment (snaiperskaya96) [inline server/methods/sendConfirmationEmail.coffee]: Oh dear :man_facepalming: fixed

**Pull request merged**